dfusion_rust_core/src/models/standing_order.rs

```diff
@@ -123,31 +128,67 @@ impl From<&Arc<Log>> for StandingOrder {
 }
 
 
-impl From<Entity> for StandingOrder {
-    fn from(entity: Entity) -> Self {
-        let account_id = u16::from_entity(&entity, "accountId");
-        let batch_index = U256::from_entity(&entity, "batchIndex");
-        let valid_from_auction_id = U256::from_entity(&entity, "validFromAuctionId");
-
-        StandingOrder {
-            account_id,
-            batch_index,
-            valid_from_auction_id,            
-            orders: vec![] // Orders cannot be reconstructed just from the entity
+impl From<(Entity, Vec<Entity>)> for StandingOrder {
+    fn from(entities: (Entity, Vec<Entity>)) -> Self {
+        let batch_entity = entities.0;
+        let mut order_entities_iter = entities.1.into_iter();
+        let order_ids_value = batch_entity.get("orders").expect("it should contain order ids");
+
+        if let Value::List(order_ids) = order_ids_value {
+            assert_eq!(
+                order_ids.len(),
+                order_entities_iter.len(),
+                "The entity should have the same number of orders as Vec<Entity>"
+            );
+
+            let account_id = u16::from_entity(&batch_entity, "accountId");
+            let batch_index = U256::from_entity(&batch_entity, "batchIndex");
+            let valid_from_auction_id = U256::from_entity(&batch_entity, "validFromAuctionId");
+
+            let orders = order_ids.iter()
+                .map(|order_id| {
+                    // Get matching order for the id
+                    let order_entity: Entity = order_entities_iter
+                        .find(|current_order| {
```

Comment:
Unnecessary O(n^2) but who cares when n is 10.

Reply:
Why do you say unnecessary:
- Because we can do it linear, i.e. using hash map
- Because we think is enough with assuming the orders match the ids and in the same order, so it's just linear
- For any other reason

I think a good principle to keep in mind is:
![image](https://user-images.githubusercontent.com/2352112/63109318-33ae5a80-bf89-11e9-951b-0975ed12d72b.png)

I don't think this will be the bottleneck in this system